src/components/chat/chat-panel.tsx

```diff
@@ -25,13 +27,13 @@ export const ChatPanel: React.FC<IProps> = ({ user, activeNavTab, focusDocument,
   const { data: comments } = useDocumentComments(focusDocument);
   const { data: unreadComments } = useUnreadDocumentComments(focusDocument);
   const documentComments = comments?.filter(comment => comment.tileId == null);
-  const allTileComments = comments?.filter(comment=> comment.tileId !== null);
+  const allTileComments = comments?.filter(comment=> comment.tileId != null);
   const commentsInDocumentOrder = ordering && allTileComments
     ? allTileComments.sort((a: any, b: any) => ordering.indexOf(a.tileId) - ordering.indexOf(b.tileId))
     : [];
 
- 
   const postedComments = documentComments?.concat(commentsInDocumentOrder);
+  const commentThreads = makeChatThreads(postedComments, content);
```

Comment:
Because this is making a new array each time, this will cause a re-render of the ChatThread component on every render of ChatPanel. Normally this would be OK. But if a user is typing in a comment and then another user provides a comment, the user that is typing might lose what they are working on.  I'm not totally sure though, but it should be easy to test with two browser tabs.

If this is a problem, I think this is something that could be punted to a new new story and PR. 

My suggested fix would be to use some "fixed" (probably need to use a useRef to do that), MobX objects for the `commentThreads`. And then update the CommentCard component so has two children `CommentList`(new) and `CommentTextBox`. So this way the CommentList could be re-renderd whenever a new comment comes in, but the CommentTextBox would not be re-rendered.

It'd get more complicated when dealing with a tile that doesn't have any comments yet.

Reply:
Nice catch. It doesn't seem to be a problem for tiles/documents that already have comments, but it is for ones that have 0. I filed this bug/story: https://www.pivotaltracker.com/n/projects/2441242/stories/183428790